libz-rs-sys/README.md

```diff
@@ -88,6 +88,31 @@ let inflated = &output[..strm.total_out as usize];
 assert_eq!(inflated, input.as_bytes())
 ```
 
+## Memory Management
+
+Zlib does not assume a global allocator, but instead accepts `zalloc` and `zfree` functions for allocating and deallocating memory.
+
+```rust,ignore
+pub type alloc_func = unsafe extern "C" fn(*mut c_void, c_uint, c_uint) -> *mut c_void;
+pub type free_func = unsafe extern "C" fn(*mut c_void, *mut c_void);
+
+struct z_stream {
+    // ...
+    pub zalloc: Option<alloc_func>,
+    pub zfree: Option<free_func>,
```

Comment:
```suggestion
type alloc_func = unsafe extern "C" fn(*mut c_void, c_uint, c_uint) -> *mut c_void;
type free_func = unsafe extern "C" fn(*mut c_void, *mut c_void);

struct z_stream {
    // ...
    zalloc: Option<alloc_func>,
    zfree: Option<free_func>,
```

The pub are not all that useful IMO and kind of implied by the fact that this is documentation. It was inconsistent with the `z_stream` type too.

Reply:
Also maybe document the `opaque` field too.